cmake/checks/check_01_cxx_features.cmake

```diff
@@ -277,7 +277,7 @@ IF(NOT DEFINED DEAL_II_WITH_CXX14 OR DEAL_II_WITH_CXX14)
     #
     # https://llvm.org/bugs/show_bug.cgi?id=16876
     #
-    PUSH_CMAKE_REQUIRED("${DEAL_II_CXX_FLAGS_DEBUG}")
+    PUSH_CMAKE_REQUIRED("-g")
```

Comment:
Without this change, I get
```
clang-6.0: error: argument unused during compilation: '-Wa,--compress-debug-sections' [-Werror,-Wunused-command-line-argument]
```

Reply:
But don't we need to pass the correct set of flags? For example, the language settings. Maybe adding `-Wno-error` would be better here?

Reply:
We push the version flag some lines above. Previously, the actual flags for me were:
```
-Werror -std=c++17 -Og -ggdb
```
and the flags in the original `clang` bug report were:
```
-std=c++1y -lstdc++ -g
```
Hence, passing `-g` is in this setup nearly the same as `-Wno-error` and close to the flags in the original report.

Reply:
Anyway, I changed it.